core/src/main/java/org/elasticsearch/search/aggregations/bucket/range/BinaryRangeAggregator.java

```diff
@@ -214,6 +223,99 @@ private int collect(int doc, long ord, long bucket, int lowBound) throws IOExcep
         protected abstract void doCollect(LeafBucketCollector sub, int doc, long bucket) throws IOException;
     }
 
+    static abstract class SortedBinaryRangeLeafCollector extends LeafBucketCollectorBase {
```

Comment:
I am wondering, given that this duplicates the range binary search from the other implementation, if we should abstract that away somewhere so any fix needed to this logic only needs to be made in one place?